.github/workflows/macosx_windows_ci.yaml

```diff
@@ -73,3 +73,89 @@ jobs:
         with:
           token: ${{secrets.CODECOV_TOKEN}} #required
           file: ./coverage.xml #optional
+  benchmark:
+    name: Performance Benchmark
+    needs: tests
+    env:
+      PYTHON: ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    defaults:
+     run:
+       # Adding -l {0} helps ensure conda can be found properly in windows.
+       shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+        python-version: ["3.12"]
+        include:
+          - env_name: pyuvdata_tests_windows
+            os: windows-latest
+          - env_name: pyuvdata_tests_mac_arm
+            os: macos-latest
+          - env_name: pyuvdata_tests
+            os: macos-13
+            python-version: "3.12"
+          - env_name: pyuvdata_tests
+            os: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+        with:
+          fetch-depth: 0
+
+      - name: Setup Minimamba
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          use-mamba: true
+          python-version: ${{ matrix.python-version }}
+          environment-file: ci/${{ matrix.env_name }}.yml
+          activate-environment: ${{ matrix.env_name }}
+          run-post: false
+
+      - name: Conda Info
+        run: |
+          conda info -a
+          conda list
+          PYVER=`python -c "import sys; print('{:d}.{:d}'.format(sys.version_info.major, sys.version_info.minor))"`
+          if [[ $PYVER != $PYTHON ]]; then
+            exit 1;
+          fi
+
+      # also install benchmark utility
+      - name: Install
+        run: |
+          pip install pytest-benchmark
+          pip install .
+
+      - name: Run benchmark
+        run: |
+          pytest --benchmark-only --benchmark-json output.json
+
+      # Download previous benchmark result from cache (if exists)
+      - name: Download previous benchmark data
+        uses: actions/cache@v4
+        with:
+          path: ./cache
+          key: ${{ runner.os }}-benchmark
+
+      # Run `github-action-benchmark` action
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          # What benchmark tool the output.txt came from
+          tool: 'pytest'
+          # Where the output from the benchmark tool is stored
+          output-file-path: output.json
+          # Where the previous data file is stored
+          external-data-json-path: ./cache/benchmark-data.json
+          # Workflow will fail when an alert happens
+          fail-on-alert: true
+          # Enable Job Summary for PRs
+          summary-always: true
+          # Enable commenting
+          comment-always: true
+          # Comment on the PR if the branch is not a fork
+          comment-on-alert: ${{ github.event.pull_request.head.repo.fork == false }}
```

Comment:
Is there a reason not to comment on Fork PRs?

Reply:
yeah there is, but I don't 100% follow it. I think it is mostly related to github pages things (you can host results on github pages with this too). So it's more a security thing against that.

Reply:
I'd be incline to let it comment on all PRs. In the past, actions were set up so they didn't run on forks and I had to go in and click a button to actually get the actions to run on fork PRs, which is annoying.